tensorflow/java/src/gen/cc/op_specs.h

```diff
@@ -94,26 +94,27 @@ class AttributeSpec {
   // jni_type: the type of this attribute in JNI layer (see OperationBuilder)
   // description: a description of this attribute, in javadoc
   // iterable: true if this attribute is a list
-  // has_default_value: true if this attribute has a default value if not set
+  // default_value: default value for this attribute or nullptr if none
```

Comment:
This is guaranteed, since those default values comes from the `OpRegistry::Global()` singleton that holds the list of the registered `OpDef`s for the whole life of the application.

As you suggested, I added a mention about it.